src/Raven.Server/Documents/Sharding/ShardedContext.cs

```diff
@@ -14,32 +24,63 @@ public unsafe class ShardedContext
     {
         public const int NumberOfShards = 1024 * 1024;
 
+        public QueryMetadataCache QueryMetadataCache = new QueryMetadataCache();
+
+        private readonly ServerStore _serverStore;
         private readonly DatabaseRecord _record;
         public RequestExecutor[] RequestExecutors;
         private readonly long _lastClientConfigurationIndex;
 
-        public ShardedContext(ServerStore server, DatabaseRecord record)
+        private readonly ConcurrentDictionary<string, IndexDefinition> _cachedMapReduceIndexDefinitions = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
move to regular Dict?

Reply:
Ping

Reply:
https://github.com/ravendb/ravendb/blob/v5.3/src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/Converters/CachingJsonConverter.cs#L10

Reply:
done